### ISSUE TITLE
Error with ComposableArchitecture Dependency

### ISSUE THREAD

**Issue by wynioux** [tuist/tuist#5578]
### What happened?

I get an error when I do tuist fetch.

**Important Note**: I am updating Tuist, but its version appears as 3.32.1.

```zsh
% tuist update
```
```zsh
Checking for updates...
There are no updates available
```
---
```zsh
% tuist local
```
```zsh
The following versions are available in the local environment:
- 3.33.0
```
---
```zsh
% tuist local 3.33.0
```
```zsh
Generating .tuist-version file with version 3.33.0
File generated at path /Users/bahadir.guder/Developer/GitHub/DeveloperSuite/.tuist-version
```
---
```zsh
% tuist version
```
```zsh
3.32.1
```

### How do we reproduce it?

1. Add dependency
```swift
let dependencies = Dependencies(
    swiftPackageManager: .init([
        .remote(
            url: "https://github.com/apple/swift-log.git",
            requirement: .upToNextMajor(from: "1.0.0")
        ),
        .remote(
            url: "https://github.com/pointfreeco/swift-composable-architecture",
            requirement: .upToNextMajor(from: "1.0.0")
        )
    ]),
    platforms: [
        .iOS
    ]
)
```
or
```swift
import PackageDescription

let package = Package(
    name: "PackageName",
    dependencies: [
        .package(
            url: "https://github.com/apple/swift-log.git",
            from: "1.0.0"
        ),
        .package(
            url: "https://github.com/pointfreeco/swift-composable-architecture",
            from: "1.0.0"
        )
    ]
)
```
```swift
let dependencies = Dependencies(
    swiftPackageManager: .init(),
    platforms: [
        .iOS
    ]
)
```
2. Use dependency
```swift
 let frameworkTarget = Target(
            name: name,
            platform: platform,
            product: .staticFramework,
            bundleId: "\(baseBundleID).\(name)",
            deploymentTarget: deploymentTarget,
            infoPlist: .file(path: .relativeToManifest("\(name)/Info.plist")),
            sources: [
                .glob(.relativeToManifest("\(name)/Sources/**/*.swift"))
            ],
            dependencies: [
                .external(name: "ComposableArchitecture")
            ],
            settings: .settings(configurations: [
                .debug(
                    name: .debug,
                    xcconfig: .relativeToRoot("Tuist/BuildConfigurations/FrameworkTarget-Debug.xcconfig")
                ),
                .release(
                    name: .release,
                    xcconfig: .relativeToRoot("Tuist/BuildConfigurations/FrameworkTarget-Release.xcconfig")
                )
            ]),
            coreDataModels: coreDataModels
        )
```

### Error log

```zsh
Resolved cache profile 'Development' from Tuist's defaults
Found an inconsistency between target destinations `[TuistGraph.Destination.iPad, TuistGraph.Destination.macWithiPadDesign, TuistGraph.Destination.iPhone]` and deployment target `macOS`
Found deployment platforms (macOS) missing corresponding destination
Found an inconsistency between target destinations `[TuistGraph.Destination.macWithiPadDesign, TuistGraph.Destination.iPhone, TuistGraph.Destination.iPad]` and deployment target `macOS`
Found deployment platforms (macOS) missing corresponding destination
Target CasePathsMacros has platform 'iOS' and product 'Swift Macro' which is an invalid or not yet supported combination.
Target CasePathsMacros has platform 'iOS' and product 'Swift Macro' which is an invalid or not yet supported combination.
Target ComposableArchitectureMacros has platform 'iOS' and product 'Swift Macro' which is an invalid or not yet supported combination.
Target ComposableArchitectureMacros has platform 'iOS' and product 'Swift Macro' which is an invalid or not yet supported combination.
Target CasePaths has platforms 'iOS' and product 'staticFramework' and depends on target CasePathsMacros of type 'Swift Macro' and platforms 'iOS' which is an invalid or not yet supported combination.
Target ComposableArchitecture has platforms 'iOS' and product 'staticFramework' and depends on target ComposableArchitectureMacros of type 'Swift Macro' and platforms 'iOS' which is an invalid or not yet supported combination.
Fatal linting issues found
Consider creating an issue using the following link: https://github.com/tuist/tuist/issues/new/choose
```

### macOS version

13.6 (22G120)

### Tuist version

3.32.1

### Xcode version

15.0.1 (15A507)

**Comment by Topwiz** [comment]
I have same issue too.

**Comment by ammerzon** [comment]
This will probably be fixed with https://github.com/tuist/tuist/pull/5583.

**Comment by mikhailChelbaev** [comment]
I would greatly appreciate it if the mentioned fix 👆 could be released ASAP, as I am currently stuck with development because I can't generate a project with Tuist.

In the meantime, is there a workaround for this problem or a way to install TCA without macros? As I understand, I can't just install TCA version prior to 1.4, as it still relies on internal dependencies that utilize macros, which I have no control over.

**Comment by ammerzon** [comment]
Hi, @mikhailChelbaev. Our workaround is to pin the following dependencies to the exact version:

```swift
.remote(url: "https://github.com/pointfreeco/swift-case-paths", requirement: .exact("1.0.0")),
.remote(url: "https://github.com/pointfreeco/swift-composable-architecture", requirement: .exact("1.2.0"))
```

**Comment by mikhailChelbaev** [comment]
It works, thank you @ammerzon

**Comment by veraposeidon** [comment]
i've had the same issue.
since tuist [v3.33.0](https://github.com/tuist/tuist/releases/tag/3.33.0) has **Integrate Swift Macros using XcodeProj-native building blocks**, you can depend on `ComposableArchitecture` use the Xcode native integration way for now. for a practical example, check out: [framework_with_swift_macro](https://github.com/tuist/tuist/tree/main/fixtures/framework_with_swift_macro).

here's a quick rundown,
1. add the package to Project like this:
```
let project = Project(
    name: "FrameworkWithSwiftMacro",
    packages: [
        .remote(url: "https://github.com/pointfreeco/swift-composable-architecture", requirement: .exact("1.4.0")),
    ],
```
2. add the dependency to Target use the package case, mark the packageType as `marco`, this indicates that the package contains a Swift Macro.
```
Target(
    ....
    dependencies: [
        .package(product: "ComposableArchitecture", type: .macro),
    ]
),
```

**Comment by ammerzon** [comment]
Hi @veraposeidon. Thank you for the hint. I'll try that. 
What do you mean by Xcode native integration exactly? Or what would be a non-Xcode native integration?

**Comment by waltflanagan** [comment]
👋 this should be resolved in `main` now that #5587 is merged
